sunpy/timeseries/sources/goes.py

```diff
@@ -222,50 +223,52 @@ def _parse_netcdf(filepath):
-                start_time_str = start_time_str.lstrip("seconds since").rstrip("UTC")
-                # Perform the time addition in UTime format to ignore leap seconds
-                times = Time(parse_time(start_time_str).utime + d["time"], format="utime")
-            elif "xrsa_flux" in d.variables:
-                xrsa = np.array(d["xrsa_flux"])
-                xrsb = np.array(d["xrsb_flux"])
-                xrsa_quality = np.array(d['xrsa_flags'])
-                xrsb_quality = np.array(d['xrsb_flags'])
-
-                start_time_str = d["time"].attrs["units"]
-                if not isinstance(start_time_str, str):
-                    # For h5netcdf<0.14
-                    start_time_str = start_time_str.astype(str)
-                start_time_str = start_time_str.lstrip("seconds since")
-                # Perform the time addition in UTime format to ignore leap seconds
-                times = Time(parse_time(start_time_str).utime + d["time"], format="utime")
-
-            else:
-                raise ValueError(f"The file {filepath} doesn't seem to be a GOES netcdf file.")
+        with h5netcdf.File(filepath, mode="r", **XRSTimeSeries._netcdf_read_kw) as h5nc:
+            header = MetaDict(OrderedDict(h5nc.attrs))
+            flux_name = h5nc.variables.get("a_flux") or h5nc.variables.get("xrsa_flux")
+            if flux_name is None:
+                raise ValueError(f"No flux data (either a_flux or xrsa_flux) found in file: {filepath}")
+            flux_name = flux_name.name
+            xrsa = np.array(h5nc[flux_name])
+            xrsb = np.array(h5nc[flux_name.replace("a", "b")])
+            xrsa_quality = np.array(h5nc[flux_name.replace("flux", "flags")])
+            xrsb_quality = np.array(h5nc[flux_name.replace("a", "b").replace("flux", "flags")])
+            start_time_str = h5nc["time"].attrs["units"]
+            # h5netcdf < 0.14 return bytes instead of a str
+            if isinstance(start_time_str, bytes):
+                start_time_str = start_time_str.decode("utf-8")
+            start_time_str = start_time_str.lstrip("seconds since").rstrip("UTC").strip()
+            times = Time(parse_time(start_time_str).unix + h5nc["time"], format="unix")
 
-        data = DataFrame({"xrsa": xrsa, "xrsb": xrsb, "xrsa_quality": xrsa_quality,
-                         "xrsb_quality": xrsb_quality}, index=times.datetime)
+        try:
+            times = times.datetime
+        except ValueError:
+            # We make an assumption here that there is only 1 leap second in the file.
+            # We do not make the assumption that the leap second occurs at the end of the file.
+            # Therefore, we need to find it:
+            # To do so, we convert the times to isot strings, use numpy to find the the leap second string,
+            # then use that to workout the index of the leap timestamp.
+            idx = np.argwhere((np.char.find(times.isot, ":60.") != -1) == True)
```

Comment:
as you only expect one index to be returned here, already make sure that it is a number and not an array of arrays

Reply:
This does not return a number but an nested array with an number. Luckily when I use it to index the original array, it handles it.

Reply:
Oh you mean, I should assert it has one value.

Reply:
This should be fixed now.

Reply:
theoretically ValueError could happen for a number of reasons (not just because of leap seconds), being a bit defensive is therefore better ;) currently it would have failed for the following statement if there was no index found `times[idx].isot.tolist()[0][0]` with your fix it should be clearer - when revisiting the code `tolist()[0][0][:17]` might still bear some mysteries for future readers that you might want to improve. e.g.

```
idx = idx[0][0]
leap_second = times[idx]
rounded_leap_second = leap_second[:17] + "59.999"
times[idx] = rounded_leap_second
```

not sure, maybe too verbose

Reply:
That is fair, at least for getting this out sooner, I probably won't make more changes. 

This code will also be deleted when we transition to astropy.times for Timeseries for 4.1 so I am ok with it being a bit rough.